src/transformers/models/detr/modeling_detr.py

```diff
@@ -520,16 +520,16 @@ def __init__(
     def _shape(self, tensor: torch.Tensor, seq_len: int, batch_size: int):
         return tensor.view(batch_size, seq_len, self.num_heads, self.head_dim).transpose(1, 2).contiguous()
 
-    def with_pos_embed(self, tensor: torch.Tensor, position_embeddings: Optional[Tensor]):
-        return tensor if position_embeddings is None else tensor + position_embeddings
+    def with_pos_embed(self, tensor: torch.Tensor, object_queries: Optional[Tensor]):
+        return tensor if object_queries is None else tensor + object_queries
 
     def forward(
         self,
         hidden_states: torch.Tensor,
         attention_mask: Optional[torch.Tensor] = None,
-        position_embeddings: Optional[torch.Tensor] = None,
+        object_queries: Optional[torch.Tensor] = None,
         key_value_states: Optional[torch.Tensor] = None,
-        key_value_position_embeddings: Optional[torch.Tensor] = None,
+        spatial_position_embeddings: Optional[torch.Tensor] = None,
         output_attentions: bool = False,
     ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
```

Comment:
For public methods, we need to maintain backwards compatibility. In this case, I'd suggest doing the following: 

```python
def forward(
    self, 
    ..., # all the current keyword args here
    # we add kwargs but add protections so it's only used for the deprecated arguments
    **kwargs
): 
    position_embeddings = kwargs.pop("position_ebmeddings", None)
    key_value_position_embeddings = kwargs.pop("key_value_position_embeddings", None)

    if kwargs:
        raise ValueError(f"Unexpected arguments {kwargs.keys()}")

    if position_embeddings is not None and object_queries is not None: 
        raise ValueError("Cannot specify both position_embeddings and object_queries. Please use just object_queries")

    if key_value_position_embeddings is not None and spatial_position_embeddings is not None: 
        raise ValueError("Cannot specify both key_value_position_embeddings and spatial_position_embeddings. Please use just spatial_position_embeddings")
    
    if position_embeddings is not None:
        logger.warning_once("position_embeddings has been deprecated and will be removed in v4.34. Please use object_queries instead")
        object_queries = position_embeddings

    if key_value_position_embeddings is not None: 
        logger.warning_once("key_value_position_embeddings has been deprecated and will be removed in v4.34. Please use spatial_position_embeddings instead")
        spatial_position_embeddings = key_value_position_embeddings
```

The equivalent will need to be applied to all the other public methods e.g. other forwards, with_pos_embed etc.

Reply:
Makes sense! Will do! Thanks for feedback